pkg/storage/replica_command.go

```diff
@@ -379,7 +384,11 @@ func (r *Replica) AdminMerge(
 		if err := r.store.DB().GetProto(ctx, rightDescKey, &rightDesc); err != nil {
 			return reply, roachpb.NewError(err)
 		}
-
+		// lhs.Generation = max(rhs.Generation, lhs.Generation)+1.
+		// See the comment on the Generation field for why generation are useful.
+		if updatedLeftDesc.GetGeneration() > rightDesc.GetGeneration() {
```

Comment:
Hold on, is this backwards? Should this be `if updatedLeftDesc.GetGeneration() < rightDesc.GetGeneration() {`?

Reply:
If so, we're probably missing a test where the RHS of a merge has a larger generation than the LHS.

Reply:
For posterity, this was fixed in https://github.com/cockroachdb/cockroach/pull/36754